### PR TITLE
Componentize summary data

### PR DESCRIPTION
**Pull request by jcoyne**
This prevents a bug where the Summary field is drawn if the catalog/_summary_data component returns anything, even whitespace.

### CODE REVIEW

File: spec/components/toc_and_summary_component_spec.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+RSpec.describe TocAndSummaryComponent, type: :component do
+  pending "add some examples to (or delete) #{__FILE__}"
+
+  # it "renders something useful" do
+  #   expect(
+  #     render_inline(described_class.new(attr: "value")) { "Hello, components!" }.css("p").to_html
+  #   ).to include(
+  #     "Hello, components!"
+  #   )
+  # end
```
Comment (marlo-longley) [inline spec/components/toc_and_summary_component_spec.rb:14]: @jcoyne should we clean up these fake test comments?

**Review by marlo-longley — CHANGES_REQUESTED** [review]
Hi @jcoyne  -- I reviewed this an still had some issues with the blank summary field -- certain examples listed here (https://github.com/sul-dlss/SearchWorks/pull/3914) were fixed ("Pariatur") but others were broken ("World 1").

Screenshot 2024-02-21 at 10 25 33 AM

Screenshot 2024-02-21 at 10 38 43 AM

At least in the case of "World 1" this is because `render_toc_and_summary?` is returning true for these records -- because `document.key?(:summary_struct)` is returning true. Here's what's in there for "World 1" -- and several others with the same persisting issue. 
 
`summary_struct [{"label"=>"Summary", "fields"=>[], "unmatched_vernacular"=>[]}, {"label"=>"Content advice", "fields"=>[], "unmatched_vernacular"=>[]}]`

I need to look closer into the sanitize helper for this field, but maybe we should exclude certain types of summary_struct data.

**Comment by jcoyne** [comment]
@marlo-longley the goal here wasn't to prevent this problem when we have bad data, but to prevent this problem when there was no data at all.

**Comment by marlo-longley** [comment]
Ah, gotcha I misunderstood the original issue. I just spotted this and thought it was related.

**Review by marlo-longley — APPROVED** [review]
@jcoyne since the issue I found is fixed in this traject PR (searchworks.stanford.edu/view/14175447) and is separate from this, I'm approving.

With the branch checked out, the records mentioned in the original issue looked fixed.

File: spec/components/summary_data_component_spec.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe SummaryDataComponent, type: :component do
+  pending "add some examples to (or delete) #{__FILE__}"
+
+  # it "renders something useful" do
```
Comment (marlo-longley) [inline spec/components/summary_data_component_spec.rb:8]: @jcoyne this needs cleanup of comments as well.

Comment (jcoyne) [reply spec/components/summary_data_component_spec.rb]: done.

**Review by marlo-longley — APPROVED** [review]

**Pull request merged**